File: Nx3CS/bin/test/src/nx3/PNoteHeadsRectsPausesCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PNoteHeadsRectsPausesCalculator : global::haxe.lang.HxObject {
		public    PNoteHeadsRectsPausesCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PNoteHeadsRectsPausesCalculator(global::nx3.PNote vnote){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				global::nx3.PNoteHeadsRectsPausesCalculator.__hx_ctor_nx3_PNoteHeadsRectsPausesCalculator(this, vnote);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PNoteHeadsRectsPausesCalculator(global::nx3.PNoteHeadsRectsPausesCalculator __temp_me108, global::nx3.PNote vnote){
			unchecked {
				#line 16 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				__temp_me108.vnote = vnote;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				return new global::nx3.PNoteHeadsRectsPausesCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				return new global::nx3.PNoteHeadsRectsPausesCalculator(((global::nx3.PNote) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PNote vnote;
		
		public virtual   global::Array<object> getHeadsRects(){
			unchecked {
				#line 22 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				global::Array<object> rects = default(global::Array<object>);
				#line 22 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				{
					#line 22 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
					int _g = global::nx3.ENoteValTools.beaminglevel(this.vnote.nnote.@value);
					#line 22 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
					switch (_g){
						case 1:
						{
							#line 24 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							rects = new global::Array<object>(new object[]{new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-1.8, true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(3.6, true), new global::haxe.lang.Null<double>(((double) (6) ), true))});
							#line 24 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							break;
						}
						
						
						case 2:
						{
							#line 25 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							rects = new global::Array<object>(new object[]{new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (-2) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (4) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true))});
							#line 25 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							break;
						}
						
						
						default:
						{
							#line 26 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							rects = new global::Array<object>(new object[]{new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-1.6, true), new global::haxe.lang.Null<double>(-3.3, true), new global::haxe.lang.Null<double>(3.2, true), new global::haxe.lang.Null<double>(6.6, true))});
							#line 26 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							break;
						}
						
					}
					
				}
				
				#line 29 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				int level = 0;
				{
					#line 30 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
					global::nx3.ENoteType _g1 = this.vnote.nnote.type;
					#line 30 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
					switch (global::Type.enumIndex(_g1)){
						case 1:
						{
							#line 30 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							int l = ((int) (global::haxe.lang.Runtime.toInt(_g1.@params[0])) );
							#line 32 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							level = l;
							#line 32 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							break;
						}
						
						
						default:
						{
							#line 30 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							{
							}
							
							#line 30 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
							break;
						}
						
					}
					
				}
				
				#line 36 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				((global::nx3.geom.Rectangle) (rects[0]) ).offset(((double) (0) ), ((double) (level) ));
				#line 38 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				return rects;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				switch (hash){
					case 978488168:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						this.vnote = ((global::nx3.PNote) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getHeadsRects") ), ((int) (325402290) ))) );
					}
					
					
					case 978488168:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return this.vnote;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return this.getHeadsRects();
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				baseArr.push("vnote");
				#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
				{
					#line 10 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsPausesCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PNoteHeadsRectsPausesCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PNoteHeadsRectsPausesCalculator extends haxe.lang.HxObject
{
	public    PNoteHeadsRectsPausesCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PNoteHeadsRectsPausesCalculator(nx3.PNote vnote)
	{
		nx3.PNoteHeadsRectsPausesCalculator.__hx_ctor_nx3_PNoteHeadsRectsPausesCalculator(this, vnote);
	}
	
	
	public static   void __hx_ctor_nx3_PNoteHeadsRectsPausesCalculator(nx3.PNoteHeadsRectsPausesCalculator __temp_me123, nx3.PNote vnote)
	{
		__temp_me123.vnote = vnote;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PNoteHeadsRectsPausesCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PNoteHeadsRectsPausesCalculator(((nx3.PNote) (arr.__get(0)) ));
	}
	
	
	public  nx3.PNote vnote;
	
	public   haxe.root.Array<nx3.geom.Rectangle> getHeadsRects()
	{
		haxe.root.Array<nx3.geom.Rectangle> rects = null;
		{
			int _g = nx3.ENoteValTools.beaminglevel(this.vnote.nnote.value);
			switch (_g)
			{
				case 1:
				{
					rects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{new nx3.geom.Rectangle(((java.lang.Object) (-1.8) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (3.6) ), ((java.lang.Object) (6) ))});
					break;
				}
				
				
				case 2:
				{
					rects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{new nx3.geom.Rectangle(((java.lang.Object) (-2) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (4) ), ((java.lang.Object) (6) ))});
					break;
				}
				
				
				default:
				{
					rects = new haxe.root.Array<nx3.geom.Rectangle>(new nx3.geom.Rectangle[]{new nx3.geom.Rectangle(((java.lang.Object) (-1.6) ), ((java.lang.Object) (-3.3) ), ((java.lang.Object) (3.2) ), ((java.lang.Object) (6.6) ))});
					break;
				}
				
			}
			
		}
		
		int level = 0;
		{
			nx3.ENoteType _g1 = this.vnote.nnote.type;
			switch (haxe.root.Type.enumIndex(_g1))
			{
				case 1:
				{
					int l = ((int) (haxe.lang.Runtime.toInt(_g1.params.__get(0))) );
					level = l;
					break;
				}
				
				
				default:
				{
					{
					}
					
					break;
				}
				
			}
			
		}
		
		rects.__get(0).offset(((double) (0) ), ((double) (level) ));
		return rects;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef725 = true;
			switch (field.hashCode())
			{
				case 112362856:
				{
					if (field.equals("vnote")) 
					{
						__temp_executeDef725 = false;
						this.vnote = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef725) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef726 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef726 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getHeadsRects"))) );
					}
					
					break;
				}
				
				
				case 112362856:
				{
					if (field.equals("vnote")) 
					{
						__temp_executeDef726 = false;
						return this.vnote;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef726) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef727 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef727 = false;
						return this.getHeadsRects();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef727) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("vnote");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


